airflow/models/variable.py

```diff
@@ -195,8 +195,7 @@ def update(
 
         if Variable.get_variable_from_secrets(key=key) is None:
             raise KeyError(f"Variable {key} does not exist")
-
-        obj = session.query(Variable).filter(Variable.key == key).first()
+        obj = session.scalars(select(Variable).where(Variable.key == key)).first()
```

Comment:
```suggestion
        obj = session.scalar(select(Variable).where(Variable.key == key))
```

Equivalent since `key` is unique.